### PR TITLE
test: TT-281 UI Test Coverage Second Part

### PR DESCRIPTION
**Pull request by bryanAguilar001**
Unit test coverage is complete. Currently the coverage is 99.35%.

### CODE REVIEW

**Comment by bryanAguilar001** [comment]
There are some duplicate lines since there are two components where the same method is tested. Since it is not a very complex block, it has been decided to omit the duplicate code warning.

File: src/app/modules/shared/components/details-fields/details-fields.component.spec.ts [hunk 1]
```diff
@@ -730,6 +729,33 @@ describe('DetailsFieldsComponent', () => {
...
+    component.openOrCloseDatePicker(datepicker);
+    expect(datepicker.close).toHaveBeenCalled();
+  });
+
+  it('call the open method if opened equals false', () => {
+    const datepicker: any = { opened : false, open : () => ({}), close : () => ({}) };
+    spyOn(datepicker, 'open');
+    component.openOrCloseDatePicker(datepicker);
+    expect(datepicker.open).toHaveBeenCalled();
+  });
```
Comment (LuisMS7) [inline src/app/modules/shared/components/details-fields/details-fields.component.spec.ts:757]: These unit tests are duplicated in the file "input-date.component.spec.ts". You should test a functionality only once, keep the tests in the file where the date picker is more relevant.
https://github.com/ioet/time-tracker-ui/blob/513e48cd8a0219b336650d6d6441b2973d52221f/src/app/modules/shared/components/input-date/input-date.component.spec.ts#L40-L52

File: src/app/modules/activities-management/store/activity-management.reducers.spec.ts [hunk 2]
```diff
@@ -142,4 +142,29 @@ describe('activityManagementReducer', () => {
...
+  });
+
+  it('on ResetActivityIdToEdit, message equal to \"Reset activityIdToEdit property\"', () => {
+    const action = new actions.ResetActivityToEdit();
+
+    const state = activityManagementReducer(initialState, action);
+
+    expect(state.message).toEqual('Reset activityIdToEdit property');
+    expect(state.isLoading).toBeFalse();
+  });
```
Comment (LuisMS7) [inline src/app/modules/activities-management/store/activity-management.reducers.spec.ts:162]: Check the break lines in the tests.

File: src/app/modules/shared/components/input-date/input-date.component.spec.ts [hunk 3]
```diff
@@ -37,6 +37,42 @@ describe('InputDateComponent', () => {
...
+    const value: any = null;
+    component.writeValue(value);
+    expect(component.value).toEqual('');
+  });
+
+  it(`value shuold be '' in writeValue function when parameter is not defined`, () => {
+    const value: any = undefined;
+    component.writeValue(value);
+    expect(component.value).toEqual('');
+  });
```
Comment (LuisMS7) [inline src/app/modules/shared/components/input-date/input-date.component.spec.ts:74]: The word "should" is badly written.

**Review by LuisMS7 — CHANGES_REQUESTED** [review]

File: src/app/modules/shared/components/details-fields/details-fields.component.spec.ts (hunk 1 above)
Comment (epulla) [reply src/app/modules/shared/components/details-fields/details-fields.component.spec.ts:757]: Luis is right. You should take a look at the checks when doing a PR. They are shown like these:
image
Please also check the [SonarCloud report](https://sonarcloud.io/component_measures?id=ioet_time-tracker-ui&pullRequest=800&metric=new_duplicated_lines_density&view=list) of this PR to see what's the problem (Code duplication) and where it is located

**Review by epulla — CHANGES_REQUESTED** [review]
Please do not forget that the tests are written using "it", so if there is a verb, you should write the tests as "it **calls**..." or "it **uses**...".

**Review by epulla — APPROVED** [review]

**Review by LuisMS7 — APPROVED** [review]

**Pull request merged**